app/services/form_response.rb

```diff
@@ -19,7 +19,7 @@ def success?
 
   def to_h
     hash = { success: success }
-    hash[:errors] = errors.presence if !serialize_error_details_only?
+    hash[:errors] = (!defined?(@error_details) && errors).presence if !serialize_error_details_only?
```

Comment:
I feel like it might be clearer if all the "test" logic was together?
```suggestion
    hash[:errors] = errors.presence if !defined?(@error_details) && !serialize_error_details_only?
```

Reply:
There's a very subtle difference in the behavior there, where the `if` will avoid assigning altogether if not satisfied, as opposed to assigning to `nil`. I think there's test coverage which should demonstrate a failure with your proposed change.

Reply:
Why would we want to assign `nil` ? And aren't we doing `.compact` to remove nils anyways?

Reply:
I'm also hoping this logic is temporary, per the "future pull requests" note in the original comment.

Reply:
> Why would we want to assign `nil` ? And aren't we doing `.compact` to remove nils anyways?

We remove `nil` in analytics, so it's disregarded in `have_logged_event`. But it still exists in specs which assert against the output of `FormResponse#to_h` (e.g. many in `spec/forms`). My hope was that keeping it assigned for now would limit the diff of this pull request, with the intention of removing it altogether in a future pull request.

But in retrospect, it still ended up with a lot of diffs changing from `errors: { ... }` to `errors: nil`.

Reply:
Ah yes, that makes sense now for keeping the diff small